src/EcTranslators/GoogleTranslator/GoogleTranslatorAutoConfigDialog.cs

```diff
@@ -483,80 +281,47 @@ private void comboBoxTargetLanguages_SelectedIndexChanged(object sender, EventAr
 
 			// we don't initialize the script language(s) based on the target language changing unless we're in one of the transliterate modes
 			string selectedItem = (string)comboBoxTargetLanguages.SelectedItem;
-			if ((TargetLanguageNameMustBeConfigure == selectedItem) ||
-				(transductionSelected == TransductionType.Translate) ||
-				(transductionSelected == TransductionType.DictionaryLookup))
+			if (TargetLanguageNameMustBeConfigure == selectedItem)
 			{
 				return;
 			}
 
-			// reinitialize these so they get updated if/whenever the target language changes
-			InitializeSourceScriptBasedOnComboBoxValuesAndTransductionType();
-
-			_targetScriptLanguagesInitialized = false;
-			InitializeTargetScriptLanguages();
+			var targetLanguage = LanguagesSupported.FirstOrDefault(l => l.Code == ExtractCode(selectedItem));
 		}
 
-		private void InitializeSourceScriptBasedOnComboBoxValuesAndTransductionType()
+		private void comboBoxSourceLanguages_SelectedIndexChanged(object sender, EventArgs e)
 		{
-			string relevantSelectedItem, possibleTargetScript = null;
-			if (transductionSelected == TransductionType.Transliterate)
-			{
-				relevantSelectedItem = (string)comboBoxSourceLanguages.SelectedItem;
-				possibleTargetScript = (string)comboBoxTargetScripts.SelectedItem;
-			}
-			else if (transductionSelected == TransductionType.TranslateWithTransliterate)
-				relevantSelectedItem = (string)comboBoxTargetLanguages.SelectedItem;
-			else
-				return;	// but only with a transliterate flavor
-
-			InitializeSourceScript(relevantSelectedItem, possibleTargetScript);
+			IsModified = true;	// modified even if we don't update script languages
 		}
 
-		private void comboBoxSourceLanguages_SelectedIndexChanged(object sender, EventArgs e)
+		private void ButtonSetGoogleTranslateApiKey_Click(object sender, EventArgs e)
 		{
-			IsModified = true;	// modified even if we don't update script languages
+#if encryptingNewCredentials
+			var googleTranslatorKeyHide = Properties.Settings.Default.GoogleTranslatorCredentials;
+			var credentials = EncryptionClass.Encrypt(googleTranslatorKeyHide);
+#endif
+			// only send the key if it's already the override key (so we don't expose ours)
+			var translatorCredentialsOverride = Properties.Settings.Default.GoogleTranslatorCredentialsOverride;
+			if (!String.IsNullOrEmpty(translatorCredentialsOverride))
+				translatorCredentialsOverride = EncryptionClass.Decrypt(translatorCredentialsOverride);
 
-			// we don't initialize the script language(s) based on the source language changing unless we're in the Transliterate mode
-			string selectedItem = (string)comboBoxSourceLanguages.SelectedItem;
-			if ((SourceLanguageNameAutoDetect == selectedItem) ||
-				(transductionSelected != TransductionType.Transliterate))
+			var dlg = new QueryForGoogleCredentials(translatorCredentialsOverride);
```

Comment:
`using (var dlg = new . . .)` to ensure `dlg` is properly disposed

Reply:
thanks. done